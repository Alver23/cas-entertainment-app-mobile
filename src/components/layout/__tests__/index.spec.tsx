// React
import React, { PropsWithChildren, ReactElement } from 'react';
import { Text } from 'react-native';
import { shallow, ShallowWrapper } from 'enzyme';

// Components
import { LayoutView } from '../index';

describe('LayoutView component', () => {
  let component: ShallowWrapper<PropsWithChildren<any>>;

  const fakeMessage = 'fake component';
  const FakeComponent = (): ReactElement => <Text>{fakeMessage}</Text>;

  beforeEach(() => {
    component = shallow(
      <LayoutView>
        <FakeComponent />
      </LayoutView>,
    );
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const { children } = component.props().children.type().props;
    expect(children).toBe(fakeMessage);
  });
});
