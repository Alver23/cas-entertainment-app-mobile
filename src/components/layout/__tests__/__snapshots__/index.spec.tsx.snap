// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutView component should save a snapshot of the component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <FakeComponent />
</View>
`;
